client/src/components/layout/CarouselSlider.tsx

```diff
@@ -0,0 +1,62 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { BsChevronCompactLeft, BsChevronCompactRight } from "react-icons/bs";
+import { products } from "../../constants/index";
+import CarouselCard from "./CarouselCard";
+
+function Caraousel() {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const prevSlide = () => {
+    const isFirstSlide = currentIndex === 0;
+    const newIndex = isFirstSlide ? products.length - 1 : currentIndex - 1;
+    setCurrentIndex(newIndex);
+  };
+
+  const nextSlide = () => {
+    const isLastSlide = currentIndex === products.length - 1;
+    const newIndex = isLastSlide ? 0 : currentIndex + 1;
+    setCurrentIndex(newIndex);
+  };
+
+  useEffect(() => {
+    const autoplay = setInterval(nextSlide, 2000); // Change slide every 2 seconds
+    return () => clearInterval(autoplay);
+  }, [currentIndex]);
+
+  const currentProduct = products[currentIndex];
+  return (
+    <div className="group relative m-auto mt-4  w-full  max-w-full">
```

Comment:
Can we try a horizontal slide animation when the cards transition?

Reply:
Will do that